### PR TITLE
PIMS-390: Dockerize Express API

### PR DESCRIPTION
**Pull request by Sharala-Perumal**
**PIMS-390:** Dockerize Express API      
                          
**Changes:**
       Created Dockerfile that starts production-ready version of API.
       Amended the docker-compose file to start the express-api container.

**Testing:**
       Run `docker compose up express-api` to get the container up and running.

## 🎯 Summary

  
[PIMS-390](https://citz-imb.atlassian.net/browse/PIMS-390)

## 🔰 Checklist

- [x] I have read and agree with the following checklist and am following the guidelines in our [Code of Conduct](CODE_OF_CONDUCT.md) document.

> - I have performed a self-review of my code.
> - I have commented my code, particularly in hard-to-understand areas.
> - I have made corresponding changes to the documentation where required.
> - I have tested my changes to the best of my ability.
> - My changes generate no new warnings.

### CODE REVIEW

File: express-api/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+RUN npm i
+
+# Add curl for health check
+RUN apt-get update
+RUN apt-get install -y curl
+
+# Copy compiled build from base
+COPY --from=base /express-api/dist .
+
+CMD [ "node", "server.js" ]
```
Comment (dbarkowsky) [inline express-api/Dockerfile:44]: Missing an end of file LF.
```suggestion
CMD [ "node", "server.js" ]

```

File: express-api/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+# Compile to JavaScript build 
+RUN npm run build
+
+#############################################
+#                 Prod Build                #
+#############################################
+FROM node:18.17.1-bullseye-slim as Prod
+
+# Set the working directory to /app
```
Comment (dbarkowsky) [inline express-api/Dockerfile:27]:
```suggestion
# Set the working directory to /express-api
```

File: express-api/.dockerignore [hunk 3]
```diff
@@ -0,0 +1,9 @@
+coverage/
+./dist
```
Comment (dbarkowsky) [inline express-api/.dockerignore:2]: Just to make it look a little more like lines 1 and 3
```suggestion
dist/
```

**Review by dbarkowsky — CHANGES_REQUESTED** [review]

File: express-api/Dockerfile (hunk 1 above)
Comment (Sharala-Perumal) [reply express-api/Dockerfile]: Included suggested change in the code.

File: express-api/Dockerfile (hunk 2 above)
Comment (Sharala-Perumal) [reply express-api/Dockerfile]: Included suggested change in the code.

File: express-api/.dockerignore (hunk 3 above)
Comment (Sharala-Perumal) [reply express-api/.dockerignore]: Included suggested change in the code.

**Review by dbarkowsky — APPROVED** [review]
Built container. Works great. 
Warned me to set the BACKEND_URL until I did. :)

**Pull request merged**